client-server-client/src/main/scala/zio/raft/client/RaftClient.scala

```diff
@@ -0,0 +1,731 @@
+package zio.raft.client
+
+import zio._
+import zio.stream._
+import zio.raft.protocol._
+import scodec.bits.{BitVector, ByteVector}
+import java.time.Instant
+import zio.raft.client.RaftClient.ZmqClientTransport
+import zio.zmq.{ZContext, ZSocket}
+import zio.raft.protocol.Codecs.{clientMessageCodec, serverMessageCodec}
+
+/** Main RaftClient implementation using pure functional state machine.
+  *
+  * The client uses a unified stream that merges all events. Each ClientState knows how to handle events and transition
+  * to new states.
+  */
+class RaftClient private (
+    zmqTransport: ZmqClientTransport,
+    config: ClientConfig,
+    actionQueue: Queue[ClientAction],
+    serverRequestQueue: Queue[ServerRequest]
+) {
+
+  def connect(): UIO[Unit] =
+    actionQueue.offer(ClientAction.Connect).unit
+
+  def submitCommand(payload: ByteVector): Task[ByteVector] =
+    for {
+      promise <- Promise.make[Throwable, ByteVector]
+      action = ClientAction.SubmitCommand(payload, promise)
+      _ <- actionQueue.offer(action)
+      result <- promise.await
+    } yield result
+
+  def disconnect(): UIO[Unit] =
+    actionQueue.offer(ClientAction.Disconnect).unit
+  
+  /** Close the client cleanly by disconnecting from the server. */
+  def close(): UIO[Unit] =
```

Comment:
this might be too slow, the stream probably terminated by now. Instead, just send the message over the transport